aws/terminator/compute.py

```diff
@@ -618,3 +618,82 @@ def created_time(self):
+
+
+class Ecs(Terminator):
+    @staticmethod
+    def create(credentials):
+        def _paginate_cluster_results(client):
+            names = client.get_paginator('list_clusters').paginate(
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['clusterArns']
+
+            if not names:
+                return []
+
+            return client.describe_clusters(name=names)['clusters']
+
+        return Terminator._create(credentials, Ecs, 'ecs', _paginate_cluster_results)
+
+    @property
+    def name(self):
+        return self.instance['clusterName']
+
+    def terminate(self):
+        def _paginate_task_results(container_instance):
+            names = self.client.get_paginator('list_tasks').paginate(
+                cluster=self.name,
+                containerInstance=container_instance,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['taskArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_taks(cluster=self.name, tasks=names)['tasks']
+
+        def _paginate_container_instance_results():
+            names = self.client.get_paginator('list_container_instances').paginate(
+                cluster=self.name,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['containerInstanceArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_container_instances(cluster=self.name, containerInstances=names)['containerInstances']
+
+        def _paginate_service_results():
+            names = self.client.get_paginator('list_services').paginate(
+                cluster=self.name,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['serviceArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_services(cluster=self.name, services=names)['services']
+
+        container_instances = _paginate_container_instance_results()
+        for name in container_instances:
+            # If there are tasks running on a container instance, stop them first
+            tasks = _paginate_task_results(name['containerInstanceArn'])
+            for task in tasks:
+                self.client.stop_task(cluster=self.name, task=task['taskArn'])
+
+            self.client.deregister_container_instance(containerInstance=name['containerInstanceArn'])
+
+        # If there are running services, delete them first
+        services = _paginate_service_results()
+        for name in services:
+            self.client.delete_service(cluster=self.name, service=name['serviceName'], force=True)
+
+        self.client.delete_cluster(cluster=self.name)
```

Comment:
We would typically avoid using waiters on the lambda to prevent timeouts, especially for anything that might take more than a minute or two. The ideal way to handle a scenario like this would be to create terminators for the cluster and any dependent resources, and increase the age limit on the cluster terminator. This is complicated by the fact that doing anything with services or container instances also requires the cluster name.